# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieCommons.h
# Commit: b668d0c685e8
# Full Hash: b668d0c685e8d0bbcd3bf5b1856389ca41be31f3
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2025-01-09 04:16:57
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 5: Exempt third-party cookie blocking if the channel is on the exception list. r=cookie-reviewers,bvandersloot,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D223384
# ==============================================================================

diff -r 08acd8256071 -r b668d0c685e8 netwerk/cookie/CookieCommons.h
--- a/netwerk/cookie/CookieCommons.h	Wed Jan 08 19:25:59 2025 +0000
+++ b/netwerk/cookie/CookieCommons.h	Wed Jan 08 19:25:59 2025 +0000
@@ -118,13 +118,12 @@
   static bool ShouldIncludeCrossSiteCookie(Cookie* aCookie,
                                            bool aPartitionForeign,
                                            bool aInPrivateBrowsing,
-                                           bool aUsingStorageAccess);
+                                           bool aUsingStorageAccess,
+                                           bool aOn3pcbException);
 
-  static bool ShouldIncludeCrossSiteCookie(int32_t aSameSiteAttr,
-                                           bool aCookiePartitioned,
-                                           bool aPartitionForeign,
-                                           bool aInPrivateBrowsing,
-                                           bool aUsingStorageAccess);
+  static bool ShouldIncludeCrossSiteCookie(
+      int32_t aSameSiteAttr, bool aCookiePartitioned, bool aPartitionForeign,
+      bool aInPrivateBrowsing, bool aUsingStorageAccess, bool aOn3pcbException);
 
   static bool IsFirstPartyPartitionedCookieWithoutCHIPS(
       Cookie* aCookie, const nsACString& aBaseDomain,